src/widgets/modals/ByProgramModal.js

```diff
@@ -0,0 +1,406 @@
+/* eslint-disable react/destructuring-assignment */
+/* eslint-disable react/forbid-prop-types */
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { StyleSheet, View } from 'react-native';
+
+import { PageContentModal } from './PageContentModal';
+
+import globalStyles, { DARK_GREY, WARM_GREY, SUSSOL_ORANGE } from '../../globalStyles';
+import { AutocompleteSelector, ToggleBar, PageButton, TextEditor } from '..';
+
+import { SETTINGS_KEYS } from '../../settings';
+import { getAllPrograms, getAllPeriodsForProgram } from '../../utilities/byProgram';
```

Comment:
byProgram should be exported in utilities/index

Reply:
done b983e71ac72281f8329e9de43dd87dbdf1ce6aac